applications/luci-app-cifsd/htdocs/luci-static/resources/view/cifsd.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+'require rpc';
+'require form';
+'require tools.widgets as widgets';
+
+var callFileStat, callFileRead, callFileWrite;
+
+callFileStat = rpc.declare({
+	object: 'file',
+	method: 'stat',
+	params: [ 'path' ],
+	expect: { '': {} }
+});
+
+callFileRead = rpc.declare({
+	object: 'file',
+	method: 'read',
+	params: [ 'path' ],
+	expect: { data: '' },
+	filter: function(s) { return (s || '').trim() }
+});
+
+callFileWrite = rpc.declare({
+	object: 'file',
+	method: 'write',
+	params: [ 'path', 'data' ]
+});
+
+return L.view.extend({
+	render: function() {
+		var m, s, o;
+
+		m = new form.Map('cifsd', _('Network Shares'));
+
+		s = m.section(form.TypedSection, 'globals');
+		s.anonymous = true;
+
+		s.tab('general',  _('General Settings'));
+		s.tab('template', _('Edit Template'));
+
+		s.taboption('general', widgets.NetworkSelect, 'interface', _('Interface'),
+			_('Listen only on the given interface or, if unspecified, on lan'));
+
+		o = s.taboption('general', form.Value, 'workgroup', _('Workgroup'));
+		o.placeholder = 'WORKGROUP';
+
+		o = s.taboption('general', form.Value, 'description', _('Description'));
+		o.placeholder = 'Cifsd on OpenWrt';
+
+		o = s.taboption('template', form.TextValue, '_tmpl',
+			_('Edit the template that is used for generating the samba configuration.'),
+			_("This is the content of the file '/etc/cifs/smb.conf.template' from which your samba configuration will be generated. \
+			Values enclosed by pipe symbols ('|') should not be changed. They get their values from the 'General Settings' tab."));
+		o.rows = 20;
+		o.cfgvalue = function(section_id) {
+			return callFileRead('/etc/cifs/smb.conf.template');
+		};
+		o.write = function(section_id, formvalue) {
+			return callFileWrite('/etc/cifs/smb.conf.template', formvalue.trim().replace(/\r\n/g, '\n') + '\n');
+		};
+
+
+		s = m.section(form.TableSection, 'share', _('Shared Directories'),
+			_('Please add directories to share. Each directory refers to a folder on a mounted device.'));
+		s.anonymous = true;
+		s.addremove = true;
+
+		s.option(form.Value, 'name', _('Name'));
+		o = s.option(form.Value, 'path', _('Path'));
+		if (callFileStat('/sbin/block') && callFileStat('/etc/config/fstab')) {
```

Comment:
Note that these functions return promises, so you cannot use them synchronously. Instead, declare a `load:` function, and have it `return Promise.all([L.resolveDefault(fs.stat('/sbin/block')), L.resolveDefault(fs.stat('/etc/config/fstab'))])`.

Then change the render function to `render: function(stat_results) { ... }` and use `if (stat_results[0] && stat_results[1])` here.